src/llmcompressor/pipelines/registry.py

```diff
@@ -0,0 +1,49 @@
+from typing import Dict, List, Optional, Tuple
+
+from loguru import logger
+
+from llmcompressor.modifiers import Modifier
+from llmcompressor.modifiers.obcq.sgpt_mixin import SparsityModifierMixin
+from llmcompressor.modifiers.quantization import GPTQModifier
+from llmcompressor.pipelines import basic, independent, layer_sequential, sequential
+from llmcompressor.typing import PipelineFn
+
+__all__ = ["PIPELINES", "get_pipeline_fn"]
+
+SEQUENTIAL_MODIFIERS = (GPTQModifier, SparsityModifierMixin)
+
+PIPELINES: Dict[str, PipelineFn] = {
+    "sequential": sequential.run_pipeline,
+    "layer_sequential": layer_sequential.run_pipeline,
+    "basic": basic.run_pipeline,
+    "independent": independent.run_pipeline,
+}
```

Comment:
If we make pipelines a class satisfying an abstract `Pipeline` base class with 
```
@abstractmethod
def run_pipeline(model:PreTrainedModel, dataloader:DataLoader):...
```
would it avoid needing maps like this or types like 
```
PipelineFn = Callable[[PreTrainedModel, torch.utils.data.DataLoader], None]
```
in the typing.py file

Reply:
I purposefully avoided adding a base class, since I think it adds more infrastructure than is required. Such a class would only have one abstract method, which imho doesn't justify a class definition.